newa/__init__.py

```diff
@@ -662,19 +668,19 @@ def generate_tf_exec_command(self, ctx: CLIContext) -> tuple[list[str], dict[str
         if not self.compose:
             raise Exception('ERROR: compose is not specified for the request')
         command += ['--compose', self.compose]
-        # process git_url
-        if not self.git_url:
-            raise Exception('ERROR: git_url is not specified for the request')
-        command += ['--git-url', self.git_url]
-        # process git_ref
-        if self.git_ref:
-            command += ['--git-ref', self.git_ref]
-        # process tmt_path
-        if self.tmt_path:
-            command += ['--path', self.tmt_path]
-        # process plan
-        if self.plan:
-            command += ['--plan', self.plan]
+        # process tmt related settings
+        if not self.tmt:
+            raise Exception('ERROR: tmt settings is not specified for the request')
+        if not self.tmt.get("url", None):
+            raise Exception('ERROR: tmt "url" is not specified for the request')
+        if self.tmt['url']:
+            command += ['--git-url', self.tmt['url']]
+        if self.tmt.get("ref") and self.tmt['ref']:
+            command += ['--git-ref', self.tmt['ref']]
+        if self.tmt.get("path") and self.tmt['path']:
+            command += ['--path', self.tmt['path']]
+        if self.tmt.get("plan") and self.tmt['plan']:
+            command += ['--plan', self.tmt['plan']]
```

Comment:
See also my note above. Once we have it, we can simply do

`command += [f"--{k}={v}" for k, v in self.tmt.items()]`

instead of lines 676-683 with a benefit to be able to pass any TF cli parameters.

Reply:
I would need to be to add e.g. `pipeline-type` for multihost part of the recipe.

Reply:
`pipeline-type` would go to the `testingfarm` section, either as a separate key or under `cli_args`. ToBeDiscussed.